Sources/MapboxNavigation/Route.swift

```diff
@@ -43,6 +43,68 @@ extension Route {
         }
     }
     
+    func congestionFeatures(legIndex: Int? = nil,
+                            isAlternativeRoute: Bool = false,
+                            roadClassesWithOverriddenCongestionLevels: Set<MapboxStreetsRoadClass>? = nil) -> [Feature] {
+        guard let coordinates = shape?.coordinates, let shape = shape else { return [] }
+        var features: [Feature] = []
+        
+        for (index, leg) in legs.enumerated() {
+            let legFeatures: [Feature]
+            
+            if let congestionLevels = leg.segmentCongestionLevels, congestionLevels.count < coordinates.count {
+                // The last coordinate of the preceding step, is shared with the first coordinate of the next step, we don't need both.
+                let legCoordinates: [CLLocationCoordinate2D] = leg.steps.enumerated().reduce([]) { allCoordinates, current in
+                    let index = current.offset
+                    let step = current.element
+                    let stepCoordinates = step.shape!.coordinates
+                    
+                    return index == 0 ? stepCoordinates : allCoordinates + stepCoordinates.suffix(from: 1)
+                }
+                
+                let mergedCongestionSegments = legCoordinates.combined(congestionLevels,
+                                                                       streetsRoadClasses: leg.streetsRoadClasses,
+                                                                       roadClassesWithOverriddenCongestionLevels: roadClassesWithOverriddenCongestionLevels)
+                
+                legFeatures = mergedCongestionSegments.map { (congestionSegment: CongestionSegment) -> Feature in
+                    var feature = Feature(LineString(congestionSegment.0))
+                    feature.properties = [
+                        CongestionAttribute: String(describing: congestionSegment.1),
+                        "isAlternativeRoute": isAlternativeRoute,
+                        CurrentLegAttribute: (legIndex != nil) ? index == legIndex : index == 0
+                    ]
+                    
+                    return feature
+                }
+            } else {
+                var feature = Feature(LineString(shape.coordinates))
+                feature.properties = [
+                    "isAlternativeRoute": isAlternativeRoute,
+                    CurrentLegAttribute: (legIndex != nil) ? index == legIndex : index == 0
+                ]
+                legFeatures = [feature]
+            }
+            
+            features.append(contentsOf: legFeatures)
+        }
+        
+        return features
+    }
+    
+    func identifier(_ routeLineType: RouteLineType) -> String {
+        let identifier = Unmanaged.passUnretained(self).toOpaque()
```

Comment:
Add a comment explaining why we need to work with unmanaged memory here. I assume it’s to ensure that each route on the map has a unique source and layer identifier. Perhaps it would be sufficient to combine `Route.routeIdentifier` with the route index?

Reply:
I think it'd be still safer to use address of an object as its identifier. `Route.routeIdentifier`s are equal for single direction calculation, and possibly route indexes depending on situation. To prevent these naming clashes I'd rather use memory address. Will add comment.